internal/client/client_interceptor.go

```diff
@@ -18,24 +18,64 @@ import (
 	"context"
 	"time"
 
+	"github.com/tikv/client-go/v2/internal/resource_control"
 	"github.com/tikv/client-go/v2/tikvrpc"
 	"github.com/tikv/client-go/v2/tikvrpc/interceptor"
+	"github.com/tikv/pd/pkg/mcs/resource_manager/client"
 )
 
 var _ Client = interceptedClient{}
 
 type interceptedClient struct {
 	Client
+	resourceControlIt client.ResourceGroupKVInterceptor
 }
 
 // NewInterceptedClient creates a Client which can execute interceptor.
 func NewInterceptedClient(client Client) Client {
-	return interceptedClient{client}
+	return interceptedClient{client, nil}
+}
+
+// SetupResourceGroupKVInterceptor sets the interceptor for resource control.
+func (r interceptedClient) SetupResourceGroupKVInterceptor(resourceControlIt client.ResourceGroupKVInterceptor) {
```

Comment:
Seems TiDB cannot know the intercepeted with the `Client` interface? I suggest use a global variable as before commit first.  my set up likes https://github.com/nolouch/tidb/commit/7b5d16e37c4776362d8a1355204a86e007ff3ce3#diff-e0ee27ac27121190407360fb0e1f4d08f2e7f86ef4584c5fcd9a6bed1f11ad3fR112-R131.

BTW, I also need a `UnetResourceGroupInterceptor` interface, then I can associate it with the switch session variable in TiDB.